### PR TITLE
Tcp win decouple

### PR DESCRIPTION
**Pull request by alfred2g**
Decouple TCP windowing from Tiny TCP + Unit Test

Description
-----------
Decouple TCP windowing from Tiny TCP 
add unit testing for both of them

Test Steps
-----------
this is mainly a test, all Unit Tests are passing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS_TCP_WIN.c [hunk 1]
```diff
@@ -217,7 +218,7 @@
 
 /**< TCP segment pool. */
     #if ( ipconfigUSE_TCP_WIN == 1 )
-        static TCPSegment_t * xTCPSegments = NULL;
+        _static TCPSegment_t * xTCPSegments = NULL;
```
Comment (AniruddhaKanhere) [inline FreeRTOS_TCP_WIN.c:221]: Why not let the script remove the `static` qualifier and then use it?
You can find this file in the Aneexed_TCP folder with all `static` qualifiers removed.

File: test/unit-test/FreeRTOS_TCP_WIN/FreeRTOS_TCP_WIN_stubs.c [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+/*void * pvPortMalloc( size_t xWantedSize ) */
+/*{ */
+/*   return malloc( xWantedSize ); */
+/*} */
+
+/*void vPortFree( void * pv ) */
+/*{ */
+/*   free( pv ); */
+/*} */
```
Comment (AniruddhaKanhere) [inline test/unit-test/FreeRTOS_TCP_WIN/FreeRTOS_TCP_WIN_stubs.c:166]: We may want to remove this commented section?

**Review by AniruddhaKanhere — COMMENTED** [review]
Hello @alfred2g,

Thanks for the PR. I just have a couple of minor comments. Would you please take a look?

Thanks

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by htibosch** [comment]
Thank you Alfred for this PR. As usual, I will try to compile it in various projects, and also run integration tests.

**Comment by htibosch** [comment]
Compilation:
There is one compilation problem when `ipconfigUSE_TCP_WIN` is zero:
"multiple definition of xTCPWindowLoggingLevel".
It is defined both in `FreeRTOS_TCP_WIN.c` and `FreeRTOS_Tiny_TCP.c`.

Testing:
All tests went well. Funny to see the influence of the TCP sliding windows on iperf: it doubles the speed of the TCP transmission.

Personally I am not enthusiastic about these cosmetic changes:
~~~diff
-    BaseType_t xIndex, xReturn;
+    BaseType_t xIndex;
+    BaseType_t xReturn;
~~~
because they introduce a lot of noise when comparing different branches. Sometime I need to compare different versions of the repo in case there are complaints from a user.
At this other hand: it does look cleaner, one declaration per code line :-)

Thanks

**Review by n9wxu — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**